### PR TITLE
Add cuda implementations for unary and binary tensor operations in #341 and #334

### PR DESCRIPTION
**Pull request by nkoppel**
This commit adds cuda kernels for the operations listed in #341 and #334. It also adds tests for huber_error and adds an "op" parameter to the binary kernels to allow huber_error to pass it's delta value to its kernel.

### CODE REVIEW

File: src/tensor_ops/exp/exp.cu [hunk 1]
```diff
@@ -10,7 +10,7 @@ extern "C" __global__ void exp_forward(
     if (i >= numel) {
         return;
     }
-    out[i] = exp(inp[i]);
+    out[i] = expf(inp[i]);
```
Comment (coreylowman) [inline src/tensor_ops/exp/exp.cu:13]: Great catch here, thanks!

File: src/tensor_ops/huber_error/mod.rs [hunk 2]
```diff
@@ -56,3 +56,34 @@ impl<S: Shape, E: Dtype, D: Device<E>, T: Tape<D>> Tensor<S, E, D, T> {
...
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        tensor::*,
+        tests::{assert_close, TestDevice},
+    };
+
+    #[test]
+    fn test_huber_error() {
```
Comment (coreylowman) [inline src/tensor_ops/huber_error/mod.rs:68]: Thanks for adding 😁

**Review by coreylowman — APPROVED** [review]
This is awesome!! Wow I was not expecting this haha what a super pleasant surprise to wake up to 😀 In another PR we can add `#[repr(C)]` to all the kernel op definitions

**Comment by nkoppel** [comment]
Thank you for the quick response! One thing which may be of concern is that my powi.cu uses the "pow" function, which works on doubles rather than floats because the powi function doesn't seem to exist in cuda c and using doubles will prevent the exponent from being rounded.  If you don't end up changing this behavior, it should be documented, as `tensor.pow(4)` may be less performant than `tensor.pow(4.0)`.

It would probably be fine to use cuda c's powf function for this, as it shouldn't make a difference unless the exponent is in the millions.

**Comment by coreylowman** [comment]
Yeah let's move to using powf and converting the exponent to a float, opening an issue for this in a sec

**Pull request merged**